src/block_iterator.rs

```diff
@@ -104,7 +118,22 @@ impl<B: BlockLike> BlockIterator<B> {
             ValueDeletable::Value(value)
         };
 
-        Some(KeyValueDeletable { key, value: v })
+        let mut key = BytesMut::from(&self.first_key[..overlap_len]);
+        key.extend_from_slice(&rest_key);
```

Comment:
same as above, be like:

```
let mut key = BytesMut::with_capacity(overlap_len + rest_key.len());
key.extend_from_slice(&self.first_key[..overlap_len]);
key.extend_from_slice(&rest_key);
```

Reply:
I have to wait until my computer is fixed. I may modify the current implementation to perform delta encoding for `prev_key`.